# Задача 27: Write-up

На сайты мы видим некую проверяющую систему, которая даёт возможность исполнить произвольный код на Python. Решение поставленной задачи ничего не даёт; на сайте есть подсказка, что задание выдаётся лишь тестовое.

Можно заметить, что stderr выполненного кода возвращается пользователю, когда ответ неверен. Это даёт возможность получать результаты произвольных команд на сервере.
Для начала следует изучить содержимое текущей директории при помощи `os.listdir()`. Видим скрипт runner.py. При его изучении выясняется, что именно он отвечает за запуск нашей программы в некоем защищённом окружении. При этом у нас есть права на запись в этот скрипт. Отредактировав его код, можно отключить защиту. Если скрипт был нечаянно повреждён, восстановить окружение можно, сбросив вариант задания.

Наконец, уже с отключённой защитой можно обнаружить каталог `/etc`, который отсутствовал в песочнице. В нём внутри — флаг.

Флаг: **ugra_use_well_tested_sandboxes_c60009116707**.
